xcm/src/v0/multi_location.rs

```diff
@@ -541,10 +541,40 @@ impl MultiLocation {
 		}
 	}
 
+	/// This function ensures a multi-junction is in it's canonicalized/normalized form, removing
+	/// any internal `[Non-Parent, Parent]` combinations.
+	pub fn canonicalize(&mut self) {
+		let mut normalized = MultiLocation::Null;
+		let mut iter = self.iter();
+		// We build up the the new normalized path by taking items from the original multi-location.
+		// When the next item we would add is `Parent`, we instead remove the last item assuming
+		// it is non-parent.
+		const EXPECT_MESSAGE: &'static str =
+			"This loop pushes at most N junctions to a new multi-location where N is the number \
```

Comment:
please stick to the house style for proofs, as documented in https://github.com/paritytech/substrate/blob/master/docs/STYLE_GUIDE.md.

this means `;` between clauses and finishing with `; qed`.